core/primitives/src/congestion_info.rs

```diff
@@ -0,0 +1,103 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use bytesize::ByteSize;
+use near_primitives_core::types::Gas;
+
+/// Stores the congestion level of a shard.
+///
+/// [`CongestionInfo`] should remain an internal struct that is not borsh
+/// serialized anywhere. This way we can change it more easily.
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CongestionInfo {
+    /// Congestion level for queued up unprocessed receipts.
+    pub incoming_congestion: u16,
+    /// Congestion level for queued up unprocessed receipts and buffered
+    /// outgoing receipts.
+    pub memory_congestion: u16,
+}
+
+/// Detailed information about receipts currently stored in the trie,
+/// potentially contributing to congestion.
+///
+/// This data is stored in the [`ChunkExtra`], therefore changing its layout
+/// requires a protocol change.
+#[derive(Default, Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct StoredReceiptsInfo {
```

Comment:
How about FullCongestionInfo or CongestionInfoExtra?

Reply:
After a discussion with bowen, it seems we need to store all information in the chunk header anyway, to allow stateless validation without downloading the previous chunk. Hence, I think this should just be `CongestionInfo`.